### PR TITLE
Home & Teacher home: show "HOC is coming" until actual HOC

### PR DESCRIPTION
**Pull request by breville**
On the homepage's call-to-action and teacher homepage's special announcement, the celebrity content was intended to coincide with "actual-hoc", but launched a week earlier.  This change updates each so that the relevant headings say "HOC is coming" until the the Hour of Code actually begins.

For the homepage, the "HOC is coming" will change to "HOC is here" when this flag is flipped (from `soon-hoc`):

```DCDO.set("hoc_mode", "actual-hoc")```

For the teacher homepage special announcement, the "HOC is coming" will change to "HOC is here" when this flag is flipped (from `celebs2017`):

```DCDO.set("hoc_launch", {"special_announcement"=>"celebs2017actualhoc"})```

#### current

screenshot 2017-11-26 23 29 21

screenshot 2017-11-26 23 32 40

#### when flags flipped

screenshot 2017-11-26 23 38 07

screenshot 2017-11-26 23 33 40

### CODE REVIEW

File: apps/src/templates/studioHomepages/TeacherHomepage.jsx [hunk 1]
```diff
@@ -94,6 +94,23 @@ export default class TeacherHomepage extends Component {
...
+            buttons={[
+              {url: pegasus('/challenge'), text: i18n.celebrityChallenge()},
+              {url: pegasus('/learn'), text: i18n.tryHOC()}
+            ]}
+          />
+        )}
+
+        {hocLaunch &&
+         hocLaunch.special_announcement &&
+         hocLaunch.special_announcement === "celebs2017actualhoc" &&
```
Comment (joshlory) [inline apps/src/templates/studioHomepages/TeacherHomepage.jsx:113]: Should this be a separate PR?  It seems like we should hotfix the "Hour of Code is here" change below, then push the announcement update as part of the normal DTP.

Comment (breville) [reply apps/src/templates/studioHomepages/TeacherHomepage.jsx:113]: Turns out that neither needs a hotfix, so we can just send them both out together as part of normal DTP.

**Comment by breville** [comment]
war approved 11/27.
